oak_restricted_kernel/src/acpi.rs

```diff
@@ -0,0 +1,363 @@
+        _segment: u16,
+        _bus: u8,
+        _device: u8,
+        _function: u8,
+        _offset: u16,
+    ) -> u32 {
+        unimplemented!()
+    }
+
+    fn write_pci_u8(
+        &self,
+        _segment: u16,
+        _bus: u8,
+        _device: u8,
+        _function: u8,
+        _offset: u16,
+        _value: u8,
+    ) {
+        unimplemented!()
+    }
+
+    fn write_pci_u16(
+        &self,
+        _segment: u16,
+        _bus: u8,
+        _device: u8,
+        _function: u8,
+        _offset: u16,
+        _value: u16,
+    ) {
+        unimplemented!()
+    }
+
+    fn write_pci_u32(
+        &self,
+        _segment: u16,
+        _bus: u8,
+        _device: u8,
+        _function: u8,
+        _offset: u16,
+        _value: u32,
+    ) {
+        unimplemented!()
+    }
+}
+
+trait TableContents<'a> {
+    fn contents(&self) -> &'a [u8];
+}
+impl<'a> TableContents<'a> for AmlTable {
+    fn contents(&self) -> &'a [u8] {
+        let virt_addr = ADDRESS_TRANSLATOR
+            .get()
+            .unwrap()
+            .translate_physical(PhysAddr::new(self.address as u64))
+            .unwrap();
+        // Safety: this address was specified in the ACPI tables by the firmware, so if the tables
+        // are correct, this is safe.
+        unsafe { core::slice::from_raw_parts(virt_addr.as_ptr(), self.length as usize) }
+    }
+}
+
+/// Translates an EISA ID (a compressed 32-bit identifier) to its string representation.
+fn name_from_eisaid(eisaid: u64) -> Result<String, AmlError> {
+    let eisaid: u32 = u32::from_be(eisaid.try_into().map_err(|_| AmlError::InvalidNameSeg)?);
+    let mut str = String::with_capacity(7);
+
+    for x in [26, 21, 16].iter() {
+        str.push(
+            char::from_u32(0x40 + ((eisaid & (0b11111u32 << x)) >> x))
+                .ok_or(AmlError::InvalidNameSeg)?,
+        );
+    }
+    for x in [12, 8, 4, 0].iter() {
+        str.push(
+            char::from_digit((eisaid & (0b1111u32 << x)) >> x, 0xF)
+                .ok_or(AmlError::InvalidNameSeg)?
+                .to_ascii_uppercase(),
+        );
+    }
+
+    Ok(str)
+}
+pub struct Acpi {
+    tables: AcpiTables<Handler>,
+    aml: AmlContext,
+}
+
+impl Acpi {
+    pub fn new(params: &BootParams) -> Result<Self> {
+        let mut acpi = Self {
+            tables: find_acpi_tables(params)?,
+            aml: AmlContext::new(Box::new(Handler {}), aml::DebugVerbosity::None),
+        };
+
+        // Parse the DSDT and all SSDTs.
+        if let Some(ref dsdt) = acpi.tables.dsdt {
+            acpi.aml
+                .parse_table(dsdt.contents())
+                .map_err(|err| anyhow!("Failed to parse ACPI DSDT: {:?}", err))?;
+        } else {
+            bail!("No DSDT found in ACPI tables");
+        }
+
+        for ssdt in &acpi.tables.ssdts {
+            acpi.aml.parse_table(ssdt.contents()).map_err(|err| {
+                anyhow!(
+                    "Failed to parse ACPI SSDT at address {}: {:?}",
```

Comment:
I believe Conrad did a global rewording of all the error messages and they are now mostly starting with lowercase